sharding-scaling nullpoint

## Bug Report
 

### Which version of ShardingSphere did you use?
4.1.0
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
sharding-scaling

### Expected behavior



### Actual behavior
I use sharding-scaling and sharding-proxy to move data from old cluster to the new cluster.
Below exception occured:
`[ERROR] 09:53:53.645 [pool-1-thread-21] o.a.s.s.c.e.e.SyncTaskExecuteCallback - RealtimeDataSyncTask realtime-hc_hiwork_suite_new execute exception exit
java.lang.NullPointerException: null
	at org.apache.shardingsphere.shardingscaling.mysql.MySQLBinlogReader.handleUpdateRowsEvent(MySQLBinlogReader.java:137)
	at org.apache.shardingsphere.shardingscaling.mysql.MySQLBinlogReader.read(MySQLBinlogReader.java:97)
	at org.apache.shardingsphere.shardingscaling.mysql.MySQLBinlogReader.run(MySQLBinlogReader.java:75)
`
the following code caused a nullpointexception:
`!oldValue.equals(newValue)`

The reason is that there is a field, it's oldValue is null.


### Reason analyze (If you can)


### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
Use scaling to sync database, and update a row which has null value. 

### Example codes for reproduce this issue (such as a github link).
The correct logic :
`boolean updated = false;
                if(null == oldValue){
                    updated = null != newValue;
                }else{
                    updated = !oldValue.equals(newValue);
                }`

--- a/sharding-scaling/sharding-scaling-mysql/src/main/java/org/apache/shardingsphere/shardingscaling/mysql/MySQLBinlogReader.java
+++ b/sharding-scaling/sharding-scaling-mysql/src/main/java/org/apache/shardingsphere/shardingscaling/mysql/MySQLBinlogReader.java
@@ -44,6 +44,7 @@ import org.apache.shardingsphere.sql.parser.binder.metadata.table.TableMetaData;
 import lombok.Setter;
 import lombok.extern.slf4j.Slf4j;
 import java.io.Serializable;
+import java.util.Objects;
 
 /**
  * MySQL binlog reader.
@@ -85,23 +86,31 @@ public final class MySQLBinlogReader extends AbstractSyncExecutor implements Log
         while (isRunning()) {
             AbstractBinlogEvent event = client.poll();
             if (null == event) {
-                try {
-                    Thread.sleep(100);
-                } catch (InterruptedException ignored) {
-                }
+                sleep();
                 continue;
             }
-            if (event instanceof WriteRowsEvent) {
-                handleWriteRowsEvent(channel, uri, (WriteRowsEvent) event);
-            } else if (event instanceof UpdateRowsEvent) {
-                handleUpdateRowsEvent(channel, uri, (UpdateRowsEvent) event);
-            } else if (event instanceof DeleteRowsEvent) {
-                handleDeleteRowsEvent(channel, uri, (DeleteRowsEvent) event);
-            } else if (event instanceof PlaceholderEvent) {
-                createPlaceholderRecord(channel, event);
-            }
+            handleEvent(channel, uri, event);
         }
         pushRecord(channel, new FinishedRecord(new NopLogPosition()));
+    }
+    
+    private void sleep() {
+        try {
+            Thread.sleep(100);
+        } catch (InterruptedException ignored) {
+        }
+    }
+    
+    private void handleEvent(final Channel channel, final JdbcUri uri, final AbstractBinlogEvent event) {
+        if (event instanceof WriteRowsEvent) {
+            handleWriteRowsEvent(channel, uri, (WriteRowsEvent) event);
+        } else if (event instanceof UpdateRowsEvent) {
+            handleUpdateRowsEvent(channel, uri, (UpdateRowsEvent) event);
+        } else if (event instanceof DeleteRowsEvent) {
+            handleDeleteRowsEvent(channel, uri, (DeleteRowsEvent) event);
+        } else if (event instanceof PlaceholderEvent) {
+            createPlaceholderRecord(channel, event);
+        }
     }
     
     private void handleWriteRowsEvent(final Channel channel, final JdbcUri uri, final WriteRowsEvent event) {
@@ -134,7 +143,7 @@ public final class MySQLBinlogReader extends AbstractSyncExecutor implements Log
             for (int j = 0; j < beforeValues.length; j++) {
                 Object oldValue = beforeValues[j];
                 Object newValue = afterValues[j];
-                record.addColumn(new Column(tableMetaData.getColumnMetaData(j).getName(), newValue, !newValue.equals(oldValue), tableMetaData.isPrimaryKey(j)));
+                record.addColumn(new Column(tableMetaData.getColumnMetaData(j).getName(), newValue, !Objects.equals(newValue, oldValue), tableMetaData.isPrimaryKey(j)));
             }
             pushRecord(channel, record);
         }

